### DISCUSSION TITLE
Reducing color depth of greyscale changes background

### DISCUSSION THREAD

**Discussion by ThoKau** [ImageMagick/ImageMagick#5667] [Digital Image Processing]
Hi there,
I have music sheet PNGs of type greyscale 16bit / 256 colors which I want to convert to 4bit PNGs. Not so difficult.
   convert   demo_gray.png  -colors 16 -depth 4  demo_gray_16col_4bit.png
generates a PNG with only 9 colors instead of 16 and a grey background instead of white.
Indentify reports "8-bit Gray 16c". What's wrong with the command?
Thanks!
![demo_gray](https://user-images.githubusercontent.com/88276594/196048358-b0703897-9423-4eaa-835c-eabb3143da49.png)
![demo_gray_16col_4bit](https://user-images.githubusercontent.com/88276594/196048367-b32ecd53-1384-417f-9c37-7c0a445f65c1.png)

**Comment by fmw42** [comment]
-colors 16 does not guarantee that many colors.  Also you are dithering if you leave off -dither none before -colors.  Try leaving that off and just use -depth 4

```
convert notes.png -depth4 x.png
convert x.png -format "%c" histogram:info:
          1181: (0,0,0) #000000 gray(0)
          1101: (17,17,17) #111111 gray(17)
           933: (34,34,34) #222222 gray(34)
           321: (51,51,51) #333333 gray(51)
           234: (68,68,68) #444444 gray(68)
           306: (85,85,85) #555555 gray(85)
           268: (102,102,102) #666666 gray(102)
           231: (119,119,119) #777777 gray(119)
           224: (136,136,136) #888888 gray(136)
           210: (153,153,153) #999999 gray(153)
           166: (170,170,170) #AAAAAA gray(170)
           137: (187,187,187) #BBBBBB gray(187)
           217: (204,204,204) #CCCCCC gray(204)
           273: (221,221,221) #DDDDDD gray(221)
         15830: (238,238,238) #EEEEEE gray(238)
          4782: (255,255,255) #FFFFFF gray(255)

```

**Comment by snibgo** [comment]
If your input contains black and white, and you want 16 colours including black and white, then the obvious operation to use is "`-posterize`", not "`-colors`".

The input is noisy, so I first use "`-level`" to make more pixels exactly black or white. Windows BAT syntax:
```
magick musicSheet.png ^
  -level 10,90%% ^
  -posterize 16 ^
  x.png
```

**Reply by ThoKau** [reply]
Hi,
thank you for your answers!
Adding  "-dither none"  and omitting "-colors 16" doesn't solve the gray background problem.
But the posterize option does! Exactly 16 colors, 4bit (and less noise around the letters compared with other sw just reducing colors).
Many thanks, problem solved!

**Reply by fmw42** [reply]
FYI. -dither none was to be put in front of -colors 16.  But I suggested removing (both).  You can get 16 colors and white and black by

`convert notes.png -depth 4 -auto-level result.png
`
But if you are satisfied with -posterize, go with that.
